M2/Macaulay2/d/convertr.d

```diff
@@ -110,96 +119,118 @@ nestingDepth(frameID:int,d:Dictionary):int := (
+       c:=convert0(w.doClause);
+       cc:=convert0(w.listClause);
+       loc:=codePosition(c);
+       when c is
+         nullCode do loc=codePosition(cc)
+	 else nothing;
+       Code(
 	  forCode(
 	       convert(w.inClause), convert(w.fromClause), convert(w.toClause),
-	       convert(w.whenClause), convert(w.listClause), 
-	       convert(w.doClause),
+	       convert(w.whenClause), unseq(cc),
+	       unseq(c),
 	       w.dictionary.frameID,
 	       w.dictionary.framesize,
-	       treePosition(e)))
-     is w:WhileDo do Code(whileDoCode(convert(w.predicate),convert(w.doClause),treePosition(e)))
-     is w:WhileList do Code(whileListCode(convert(w.predicate),convert(w.listClause),treePosition(e)))
-     is w:WhileListDo do Code(whileListDoCode(convert(w.predicate),convert(w.listClause),convert(w.doClause),treePosition(e)))
+	       combinePosition(w.forToken.position,loc)
+	  )))
+     is w:WhileDo do Code(whileDoCode(convert(w.predicate),unseq(c:=convert0(w.doClause)),combinePosition(w.whileToken.position,codePosition(c))))
+     is w:WhileList do Code(whileListCode(convert(w.predicate),unseq(c:=convert0(w.listClause)),combinePosition(w.whileToken.position,codePosition(c))))
+     is w:WhileListDo do Code(whileListDoCode(convert(w.predicate),convert(w.listClause),unseq(c:=convert0(w.doClause)),combinePosition(w.whileToken.position,codePosition(c))))
      is n:New do (
 	  if n.newparent == dummyTree
 	  then if n.newinitializer == dummyTree
-	       then Code(newCode(convert(n.newclass),treePosition(e)))
-	       else Code(newFromCode(convert(n.newclass),convert(n.newinitializer),treePosition(e)))
+	       then Code(newCode(unseq(c:=convert0(n.newclass)),combinePosition(n.newtoken.position,codePosition(c))))
+	       else Code(newFromCode(convert(n.newclass),unseq(c:=convert0(n.newinitializer)),combinePosition(n.newtoken.position,codePosition(c))))
 	  else if n.newinitializer == dummyTree
-	       then Code(newOfCode(convert(n.newclass),convert(n.newparent),treePosition(e)))
-	       else Code(newOfFromCode(convert(n.newclass),convert(n.newparent),convert(n.newinitializer),treePosition(e))))
-     is i:IfThen do Code(ifCode(convert(i.predicate),convert(i.thenclause),NullCode,treePosition(e)))
-     is i:IfThenElse do Code(ifCode(convert(i.predicate),convert(i.thenclause),convert(i.elseClause),treePosition(e)))
+	       then Code(newOfCode(convert(n.newclass),unseq(c:=convert0(n.newparent)),combinePosition(n.newtoken.position,codePosition(c))))
+	       else Code(newOfFromCode(convert(n.newclass),convert(n.newparent),unseq(c:=convert0(n.newinitializer)),combinePosition(n.newtoken.position,codePosition(c)))))
+     is i:IfThen do Code(ifCode(convert(i.predicate),unseq(c:=convert0(i.thenClause)),NullCode,combinePosition(i.ifToken.position,codePosition(c))))
+     is i:IfThenElse do Code(ifCode(convert(i.predicate),convert(i.thenClause),unseq(c:=convert0(i.elseClause)),combinePosition(i.ifToken.position,codePosition(c))))
```

Comment:
@pzinn what was the motivation behind `unseq` and the choices for `combinePosition` instead of `treePosition`? The positions seem to be random in practice, and I'm thinking about undoing some of these.